### PR TITLE
Add a warning in case new networks extend the network configuration

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Add a warning in case new networks extend the network configuration

Screen Shot 2022-05-10 at 14 11 48

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
Add a warning to configure workers dialog in case additional zones require additional network configuration
```

### CODE REVIEW

File: frontend/src/components/ShootWorkers/ManageWorkers.vue [hunk 1]
```diff
@@ -55,6 +55,15 @@ SPDX-License-Identifier: Apache-2.0
         </v-btn>
       </v-col>
     </v-row>
+    <v-row key="networkWarning" class="list-item" ref="networkWarningRef">
+      <v-alert type="warning" tile prominent v-if="hasAdditionalZonesNetworkConfiguration" class="d-flex flex-grow-1">
+        Adding addtional zones will extend the zone network configuration by adding new networks to your cluster:
+        <pre>
+<code>{{additionalZonesNetworkConfiguration}}</code>
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/ManageWorkers.vue:62]: Screenshot 2022-05-03 at 19 13 09
I'd remove the spacing between the lines

Comment (grolu) [reply frontend/src/components/ShootWorkers/ManageWorkers.vue]: done

Comment (grolu) [reply frontend/src/components/ShootWorkers/ManageWorkers.vue]: use code block

Comment (grolu) [reply frontend/src/components/ShootWorkers/ManageWorkers.vue]: use code block

Comment (grolu) [reply frontend/src/components/ShootWorkers/ManageWorkers.vue]: do not scroll

Comment (grolu) [reply frontend/src/components/ShootWorkers/ManageWorkers.vue]: do not scroll

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

File: frontend/src/components/ShootWorkers/WorkerConfiguration.vue [hunk 2]
```diff
@@ -95,6 +124,21 @@ export default {
...
+      this.additionalZonesNetworkConfiguration = []
+    }
+  },
+  watch: {
+    async hasAdditionalZonesNetworkConfiguration (value) {
+      if (value) {
+        this.additionalZonesNetworkConfigurationYaml = await this.$yaml.safeDump(this.additionalZonesNetworkConfiguration)
+      } else {
+        this.additionalZonesNetworkConfigurationYaml = undefined
+      }
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/WorkerConfiguration.vue:141]:
```suggestion
    },
    async setNetworkConfiguration (value) {
      if (value) {
        this.networkConfiguration = value
        this.networkConfigurationYaml = await this.$yaml.safeDump(value)
      } else {
        this.networkConfiguration = []
        this.networkConfigurationYaml
```
```

File: frontend/src/components/ShootWorkers/WorkerConfiguration.vue [hunk 3]
```diff
@@ -39,18 +60,26 @@ export default {
...
   },
   mixins: [
     shootItem,
     asyncRef('manageWorkers')
   ],
+  computed: {
+    hasAdditionalZonesNetworkConfiguration () {
+      return this.additionalZonesNetworkConfiguration.length
+    }
+  },
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/WorkerConfiguration.vue:82]:
```suggestion
```

File: frontend/src/components/ShootWorkers/WorkerConfiguration.vue [hunk 4]
```diff
@@ -17,15 +17,36 @@ SPDX-License-Identifier: Apache-2.0
     <template v-slot:actionComponent>
       <manage-workers
         @valid="onWorkersValid"
+        @additionalZonesNetworkConfiguration="onAdditionalZonesNetworkConfiguration"
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/WorkerConfiguration.vue:20]:
```suggestion
        @additionalZonesNetworkConfiguration="setNetworkConfiguration"
```

File: frontend/src/components/ShootWorkers/ManageWorkers.vue [hunk 5]
```diff
@@ -158,12 +159,18 @@ export default {
...
       const cloudProfile = this.cloudProfileByName(this.cloudProfileName)
-      return cloudProfile.metadata.cloudProviderKind
+      return get(cloudProfile, 'metadata.cloudProviderKind')
     },
     currentZonesNetworkConfiguration () {
       return getZonesNetworkConfiguration(this.zonesNetworkConfiguration, this.internalWorkers, this.cloudProviderKind, this.allZones.length, this.existingWorkerCIDR)
     }
   },
+  watch: {
+    currentZonesNetworkConfiguration (value) {
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ManageWorkers.vue:169]: I general we should use getter and setter and handle the dependent Data there if possible.

**Pull request merged**